lib/internal/Magento/Framework/Mail/Template/TransportBuilder.php

```diff
@@ -172,6 +172,20 @@ public function setFrom($from)
     }
 
     /**
+     * Set mail from address by store.
+     *
+     * @param string|array $from
+     * @param string|int $store
+     * @return $this
+     */
+    public function setFromByStore($from, $store)
```

Comment:
It is not encouraged to add new public methods in classes marked as api

Reply:
@ishakhsuvarov, fixed. I created separate class. Notify me if you accept these changes, i'll squash commits.

Reply:
This is a hack. Surely the correct decision should have been to mark the old setFrom function as deprecated and introduce a new working function, setFromByStore in the API.

The resulting fix that was implemented was not good, involving making objects static that shouldnt have been static, to allow other classes access its protected members. This is far worse in my opinion and has already lead to two further issues being raised caused by this commit. Can this not be looked at again and fixed properly instead of the hack that is currently in place to avoid changing the transportBuilder class.